src/indexer/services/substrateIndexer/substrateIndexer.ts

```diff
@@ -63,12 +137,64 @@ export class SubstrateIndexer {
     let latestBlock = await this.indexPastEvents()
     await this.provider.rpc.chain.subscribeNewHeads(async header => {
       // start at last block from past events query and move to new blocks range
-      if (latestBlock + this.currentEventsQueryInterval === Number(header.number)) {
+      if (latestBlock + this.currentEventsQueryInterval < Number(header.number)) {
         // connect executions to deposits
         try {
           // fetch and decode logs
+          const blockHash = await this.provider.rpc.chain.getBlockHash(latestBlock)
+          const signedBlock = await this.provider.rpc.chain.getBlock(blockHash)
+          const at = await this.provider.at(blockHash)
+          const timestamp = Number(await at.query.timestamp.now().toString())
+          const allRecords = await at.query.system.events()
+
+          // we get the proposal execution events - ts-ignore because of allRecords
+          // @ts-ignore
+          const proposalExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.ProposalExecution, allRecords) as Array<ProposalExecutionEvent>
+
+          // we get the deposit events - ts-ignore because of allRecords
+          // @ts-ignore
+          const depositEvents = getSubstrateEvents(SygmaPalleteEvents.Deposit, allRecords) as Array<DepositEvent>
+
+          // we get the index of the section in the extrinsic
+          const sectionIndex = signedBlock.block.extrinsics.findIndex(ex => ex.method.section === "sygmaBridge")
+
+          // this is our identifier for the tx
+          const txIdentifier = `${latestBlock}-${sectionIndex}` //this is like the txHash but for the substrate
+
+          if (proposalExecutionEvents.length) {
+            proposalExecutionEvents.forEach((proposalExecutionEvent: ProposalExecutionEvent) => {
+              const { data } = (proposalExecutionEvent.event as any).toHuman()
+
+              const { originDomainId, depositNonce } = data
+
+              this.saveProposalExecutionToDb(this.domain.id, latestBlock.toString(), {
+                originDomainId,
+                depositNonce: depositNonce,
+                txIdentifier,
+                blockNumber: `${latestBlock}`,
+              })
+            })
+          } else if (depositEvents.length) {
+            depositEvents.forEach((depositEvent: DepositEvent) => {
+              const { data } = (depositEvent.event as any).toHuman()
+
+              const { destDomainId, resourceId, depositNonce, sender, transferType, depositData, handlerResponse } = data
+
+              this.saveDepositToDb(this.domain.id, latestBlock.toString(), {
+                destDomainId,
+                resourceId,
+                depositNonce: depositNonce,
+                sender,
+                transferType,
+                depositData,
+                handlerResponse,
+                txIdentifier,
+                blockNumber: `${latestBlock}`,
+                timestamp
+              })
+            })
+          }
 
```

Comment:
You should handle also the `FailedHandlerExecution` event

Reply:
done